grooming code

amites opened the pull request:
This PR closes n/a

## What does this PR do?
Grooming code structure while creating docs

## How does this PR make you feel? [:link:](http://giphy.com/categories/emotions/)

![good]

jollyjerr reviewed (changes requested):
<img width="967" alt="Screen Shot 2021-07-16 at 7 47 34 PM" src="https://user-images.githubusercontent.com/51095001/126021725-8a88df3f-3f88-4962-8902-a8d8a74dfa4b.png">

Looks like an import got messed up somewhere. This is also a **lot** to test - what if we break apart the "small" stuff (typos and stuff) from the bigger stuff like adding awaits and deleting larger chunks of code. We want to make sure that everything behaves exactly as it is - and we don't really have a lot of resources to do a full regression test right now 😅. I love what is going on in here but I think taking it in smaller steps might be best? Jacob may disagree though and that's totally fair - I'm good with anything?

amites commented:
removed all the logic changes adding `await` commands and left all the pure "grooming" 

added a bunch of comments to help simplify review -- let me know if ya need to break this down further
